cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/mongo/impl/PublicationSearchProviderImpl.java

```diff
@@ -0,0 +1,60 @@
+package edu.asu.diging.cord19.explorer.core.mongo.impl;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.TextCriteria;
+import org.springframework.data.mongodb.core.query.TextQuery;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.cord19.explorer.core.model.impl.PublicationImpl;
+import edu.asu.diging.cord19.explorer.web.model.SearchType;
+
+@Service
+public class PublicationSearchProviderImpl implements SearchProvider {
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Override
+    public long getTotalResults(String title) {
+        Query query = buildQuery(title);
+        return mongoTemplate.count(query, PublicationImpl.class);
+
+    }
+
+    /**
+     * Returns a {@code List<PublicationImpl>} instance representing the publication
+     * titles matched with the input parameter 'title' .
+     * 
+     * @param title       Metadata title to be queried
+     * 
+     * @param currentPage index of current page starting from 0
+     * 
+     * @param size        length of result sublist
+     * 
+     * @return List of PublicationImpl instances containing the 'title' string
+     * 
+     **/
+    @Override
+    public List<PublicationImpl> search(String title, Long currentPage, Integer size) {
+        long startItem = currentPage * size;
+
+        Query query = buildQuery(title).sortByScore().skip(startItem).limit(size);
```

Comment:
This will probably create perfomance issues in production where we ahve over 200000 documents. `skip` still iterates over all documents (so for page 5, it first iterates over all elements of pages 1-4). Using streams seems to be the better option https://stackoverflow.com/questions/11046105/iterate-over-large-collection-in-mongodb-via-spring-data.